src/reducers/AppReducer.js

```diff
@@ -1,27 +1,25 @@
 import { Map } from 'immutable';
 
-
-
 const INITIAL_STATE = Map({
   'searchBoxOnFocus': false,
   'isModal_zipcode_shown': true,
   'isModal_post_shown': false,
+  'isModal_signIn_shown': false,
```

Comment:
Can you reduce this down to 1 field as well?

`modal_shown: "zipcode" | "signIn" | ... | null`